org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/index/IndexServiceClient.xtend

```diff
@@ -20,27 +21,37 @@ import org.eclipse.xtext.scoping.IScope
 import org.eclipse.xtext.scoping.impl.SimpleScope
 import org.slf4j.LoggerFactory
 
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION
+
 import static extension java.util.Objects.requireNonNull
+import com.google.inject.Provider
+import javax.ws.rs.client.WebTarget
 
 class IndexServiceClient implements IGlobalScopeProvider {
 
 	static val logger = LoggerFactory.getLogger(IndexServiceClient)
+	
 
 	val Client client
 	val URI baseURI
+	
+	var Provider<HttpServletRequest> requestProvider
```

Comment:
could be val, too, since it is set in the c-tor (no need to change if it is overwritten by field-injection pull request later on, right?)

Reply:
Exactly, see 113a4f5d9779c1c5a746ca41761d34ea6c80f4ed